modular_skyrat/modules/conveyor_sorter/code/conveyor_sorter.dm

```diff
@@ -63,63 +71,69 @@
 	COOLDOWN_DECLARE(use_cooldown)
 
 	light_range = 3
 	light_color = COLOR_RED_LIGHT
 
-/obj/effect/decal/cleanable/conveyor_sorter/Destroy()
+/obj/effect/decal/conveyor_sorter/Initialize(mapload)
+	. = ..()
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/effect/decal/conveyor_sorter/Destroy()
 	if(parent_item)
 		parent_item.spawned_sorters -= src
 		parent_item = null
 	return ..()
 
-/obj/effect/decal/cleanable/conveyor_sorter/examine(mob/user)
+/obj/effect/decal/conveyor_sorter/examine(mob/user)
 	. = ..()
 	. += span_notice("Attack with conveyor sorter lister to set the sorting list.")
 	. += span_notice("Slap with empty hands to change the sorting direction.")
 	. += span_notice("Alt-Click to reset the sorting list.")
 	. += span_notice("Ctrl-Click to remove.")
 
-/obj/effect/decal/cleanable/conveyor_sorter/attack_hand(mob/living/user, list/modifiers)
-	var/user_choice = tgui_input_list(user, "Choose which direction to sort to!", "Direction choice", list("North", "East", "South", "West"))
+/obj/effect/decal/conveyor_sorter/attack_hand(mob/living/user, list/modifiers)
+	var/user_choice = tgui_input_list(user, "Choose which direction to sort to!", "Direction choice", directions)
 	if(!user_choice)
 		return ..()
-	switch(user_choice)
-		if("North")
-			setDir(NORTH)
-		if("East")
-			setDir(EAST)
-		if("South")
-			setDir(SOUTH)
-		if("West")
-			setDir(WEST)
+
+	var/dir = text2dir(user_choice)
+	if(!dir)
+		return ..()
+
+	setDir(dir)
+
 	visible_message("[src] pings, updating its sorting direction!")
 	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
 
-/obj/effect/decal/cleanable/conveyor_sorter/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/conveyor_sorter))
-		var/obj/item/conveyor_sorter/cs_item = W
+/obj/effect/decal/conveyor_sorter/attackby(obj/item/used_item, mob/user, params)
+	if(istype(used_item, /obj/item/conveyor_sorter) || istype(used_item, /obj/item/conveyor_sorter/improved))
+		var/obj/item/conveyor_sorter/cs_item = used_item
 		sorting_list = cs_item.current_sort
 		visible_message("[src] pings, updating its sorting list!")
 		playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
 		return
 	else
 		return ..()
 
-/obj/effect/decal/cleanable/conveyor_sorter/AltClick(mob/user)
+/obj/effect/decal/conveyor_sorter/AltClick(mob/user)
 	visible_message("[src] pings, resetting its sorting list!")
 	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
 	sorting_list = list()
 
-/obj/effect/decal/cleanable/conveyor_sorter/CtrlClick(mob/user)
+/obj/effect/decal/conveyor_sorter/CtrlClick(mob/user)
 	visible_message("[src] begins to ping violently!")
 	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
 	qdel(src)
 
-/obj/effect/decal/cleanable/conveyor_sorter/on_entered(datum/source, atom/movable/AM)
-	. = ..()
+/obj/effect/decal/conveyor_sorter/proc/on_entered(datum/source, atom/movable/AM)
```

Comment:
Since I'm already requesting changes: Here, the `AM`, can you make it like, `entering_atom` or something like that?
<3